src/index.js

```diff
@@ -8,26 +8,48 @@ export default function ({types: t}) {
   return {
     visitor: {
       Class(path, state) {
-        if(isReactComponentClass(path)) {
-          const className = path.node.id.name;
-          injectReactDocgenInfo(className, path, state, this.file.code, t);
+        if(!isReactComponentClass(path)) {
+          return;
         }
+        const className = path.node.id.name;
+
+        if(!isExported(path, className, t)){
+          return;
+        }
+        injectReactDocgenInfo(className, path, state, this.file.code, t);
       },
       'FunctionDeclaration|FunctionExpression|ArrowFunctionExpression'(path, state) {
-        if(isStatelessComponent(path)) {
-          let className = '';
-          if(path.parentPath.node.id) {
-            className = path.parentPath.node.id.name;
-          } else {
-            return;
-          }
-          injectReactDocgenInfo(className, path, state, this.file.code, t);
+        if(!isStatelessComponent(path)) {
+          return;
+        }
+        if(!path.parentPath.node.id) {
+          return;
+        }
+        const className = path.parentPath.node.id.name;
+
+        if(!isExported(path, className, t)) {
+          return;
         }
+        injectReactDocgenInfo(className, path, state, this.file.code, t);
       },
     }
   };
 }
 
+function isExported(path, className, t){
+  const types = [
+    'ExportDeclaration',
+    'ExportNamedDeclaration',
+    'ExportDefaultDeclaration'
```

Comment:
React docgen will process the file if there is only one named declaration(multiple named declarations fail). So we have to check for that too.